### PR TITLE
feat: follow the system theme

### PR DESCRIPTION
**Pull request by BenjaminHalko**
fixes: #1260

### CODE REVIEW

**Comment by BenjaminHalko** [comment]
https://github.com/ReVanced/revanced-manager/assets/73490201/b81bb9aa-ebb4-46bd-81b1-7e84275f7a1f

**Comment by oSumAtrIX** [comment]
Wondering if this is following correct M3 specs.

PS: Banger wallpaper

**Comment by Ushie** [comment]
Instead of a DropDown button, you can use a FilledTonalButton and a popup, it looks much better

**Comment by BenjaminHalko** [comment]
@Ushie I am assuming a popup of radio buttons?

**Comment by Ushie** [comment]
yep!

**Comment by BenjaminHalko** [comment]
@Ushie Something like this:
Screenshot_20230920-111849_ReVanced Manager Debug

**Comment by Ushie** [comment]
It needs to have an apply and cancel button, and you could add an icon to the popup title maybe

This is how the old revanced manager codebase had it 
SmartSelect_20230920_212207_Hyperion

**Comment by BenjaminHalko** [comment]
@Ushie Does it really need a cancel button though? Who is going to want to cancel their theme change...

**Comment by Ushie** [comment]
I believe according to m3 guidelines the only time you provide the user with a single action is when it's an acknowledgement popup, not an action

**Comment by Ushie** [comment]
It's a material design icon, I believe that specific one is Icon.palette, you can use it like other icons are used across the app

**Comment by PalmDevs** [comment]
Yep, exactly as what Ushie said. Make sure the buttons are all text buttons though. No need for a filled button.

**Comment by Ushie** [comment]
@PalmDevs Should this dialog include a title icon? no other dialog in the app does, it may be better to not include one just for the sake of consistency at the moment

**Comment by PalmDevs** [comment]
> @PalmDevs Should this dialog include a title icon? no other dialog in the app does, it may be better to not include one just for the sake of consistency at the moment 

Technically, all dialogs should have an icon; but if no dialogs have it, don't include it for this one specific dialog.

**Comment by Ushie** [comment]
Generally the current popups in the app aren't consistent in any way so we could have one PR refactoring all the popups, I could also probably pair that with a settings refactor because settings are currently in a horrible state

File: assets/i18n/en_US.json
Comment (PalmDevs) [inline assets/i18n/en_US.json:1]: All dialogs in M3 use `Sentence case`, so change this to `App theme`.

File: assets/i18n/en_US.json [hunk 1]
```diff
@@ -178,8 +178,11 @@
         "exportSectionTitle": "Import & export",
         "logsSectionTitle": "Logs",
 
+        "themeModeLabel": "App Theme",
+        "themeModeHint": "Select the theme to use",
```
Comment (PalmDevs) [inline assets/i18n/en_US.json:182]: This is not required. Don't include any supporting text at all.

File: assets/i18n/en_US.json [hunk 2]
```diff
@@ -178,8 +178,11 @@
         "exportSectionTitle": "Import & export",
         "logsSectionTitle": "Logs",
 
+        "themeModeLabel": "App Theme",
```
Comment (PalmDevs) [inline assets/i18n/en_US.json:181]: All dialog headlines in M3 use `Sentence case`, so change this to `App theme`.

**Review by PalmDevs — CHANGES_REQUESTED** [review]
The design should ideally look exactly like what Ushie's sent in the screenshot. Except only use text buttons, filled button isn't necessary. Also needs a few string changes.

**Comment by BenjaminHalko** [comment]
Why not filled buttons, the other dialogs have them.

**Comment by PalmDevs** [comment]
> Why not filled buttons, the other dialogs have them.

We don't need to point the user to click on one specific option as clicking on both does no harm.

**Comment by BenjaminHalko** [comment]
how's this:
Screenshot_20230920-114831_ReVanced Manager Debug

**Comment by BenjaminHalko** [comment]
also, pressing back on the navbar acts like the user has pressed ok, because chances are they would rather keep the changed theme than revert back.

**Comment by BenjaminHalko** [comment]
Honestly I think that it should have a subtitle as everything else has one, or is it just me....

**Comment by BenjaminHalko** [comment]
Also, I think the ok button should be filled for consistency but again, it might be just me...

**Comment by Ushie** [comment]
> also, pressing back on the navbar acts like the user has pressed ok, because chances are they would rather keep the changed theme than revert back.

that's unexpected behavior, we shouldn't stray away from what's the norm

the only time changes are applied is with the confirm button, anything else is a dismissal

**Comment by BenjaminHalko** [comment]
@Ushie fair, though I think when `DynamicTheme` is changed it is saved automatically. (separate from the individual material you and light/dark setting) so I will have to find out how to stop it from auto saving...

**Comment by Ushie** [comment]
I don't think I understand

**Comment by BenjaminHalko** [comment]
I disabled the line that saves the theme:
```
Future<void> setThemeMode(BuildContext context, int value) async {
    //await _managerAPI.setThemeMode(value);
    final bool isDynamicTheme = DynamicTheme.of(context)!.themeId.isEven;
    await DynamicTheme.of(context)!.setTheme(value * 2 + (isDynamicTheme ? 0 : 1));
    final bool isLight = value != 2 && (value == 1 || DynamicTheme.of(context)!.theme.brightness == Brightness.light);
    SystemChrome.setSystemUIOverlayStyle(
      SystemUiOverlayStyle(
        systemNavigationBarIconBrightness:
        isLight ? Brightness.dark : Brightness.light,
      ),
    );
    notifyListeners();
  }
  ```
  And you can see that the button displaying the current theme doesn't change from System, yet it still remembers the theme.
https://github.com/ReVanced/revanced-manager/assets/73490201/1dab0d6a-bdf0-42a6-b3e6-701b123b99af

**Comment by BenjaminHalko** [comment]
@Ushie If I tell it to revert the theme on back button press, will it still matter that technically if you exit out of the app in the theme dialog it will retain your new theme despite not saving?

**Comment by Ushie** [comment]
Are you doing that to be able to live-preview? you should make it so that it applies it on confirm, no need to apply it when a radio button is clicked

**Comment by BenjaminHalko** [comment]
Ok then, sure thing!

**Comment by BenjaminHalko** [comment]
@Ushie done.

I still think the OK button should be filled.

**Comment by Ushie** [comment]
Sure, you can do that to make it consistent with other popups for the time being

**Comment by Ushie** [comment]
Could you remove the weird padding around the radio button list? I hate how android does that, you can remove it the same way I removed it in the [installer type dialog](https://github.com/ReVanced/revanced-manager/pull/1062/files#diff-6ef035c5b3abb583a047adaa0263fce17ed343f3a871d60aab2beac59f076ef1R171)

**Comment by BenjaminHalko** [comment]
@Ushie is this better?
Screenshot_20230920-144510_ReVanced Manager Debug

**Comment by Ushie** [comment]
Looks better to me, @PalmDevs what do you think?

**Comment by PalmDevs** [comment]
> @Ushie is this better?
> Screenshot_20230920-144510_ReVanced Manager Debug
> 

Can you decrease the gap between the radio button and the text to be equal to the padding of the dialog?

**Comment by BenjaminHalko** [comment]
@PalmDevs do you know how to decrease that padding?

**Comment by BenjaminHalko** [comment]
I searched it up and apparently there is no way to decrease that padding....

**Comment by BenjaminHalko** [comment]
We can add padding to the others so it balances out.....

**Comment by Ushie** [comment]
I think it'll be fine like this, the install type dialog also looks like this

**Comment by BenjaminHalko** [comment]
I do fine it odd that the space on the left of the radio button is slightly less that to the right between the text...

**Comment by BenjaminHalko** [comment]
This looks better to me personally:
Screenshot_20230920-163303_ReVanced Manager Debug

**Comment by Ushie** [comment]
If you go with that instead, you should also update the padding on the installer one

**Comment by BenjaminHalko** [comment]
I agree, if we do that should I open a new PR or just include it in this

**Comment by Ushie** [comment]
Hm, I just remembered that installer dialog includes a title icon actually

**Comment by Ushie** [comment]
Try removing the padding and adding the title icon we mentioned earlier in this thread, it might look better then

**Comment by BenjaminHalko** [comment]
`Icons.palette` gives this icon which is different than the one shown in the picture you sent, but it still works.

Screenshot_20230920-163946_ReVanced Manager Debug

**Comment by Ushie** [comment]
Ah no, I meant remove the extra padding you added in your last preview

**Comment by BenjaminHalko** [comment]
Screenshot_20230920-164453_ReVanced Manager Debug

Though I personally like this one better as the padding between the edges and text are the same:
Screenshot_20230920-164658_ReVanced Manager Debug

**Comment by BenjaminHalko** [comment]
There is also this problem we have to take into account:
Screenshot_20230920-164846_ReVanced Manager Debug

**Comment by Ushie** [comment]
Alright, you can update the installer one aswell and push your changes

**Comment by BenjaminHalko** [comment]
Sounds good

**Comment by Ushie** [comment]
> There is also this problem we have to take into account: Screenshot_20230920-164846_ReVanced Manager Debug

Hm, I think radiolists in dialogs are meant to be scrollable, let me show you an example

I'm not sure how to do that in Flutter though

**Comment by BenjaminHalko** [comment]
I think using `SingleChildScrollView`

**Comment by Ushie** [comment]
unnamed (4)

**Comment by BenjaminHalko** [comment]
fixed:
Screenshot_20230920-165221_ReVanced Manager Debug

**Comment by Ushie** [comment]
Looks good to me, since that change should also be done in installer type's dialog you should push the changes relevant to theme picker only and a followup commit in the dev branch for installer type

**Review by Ushie — APPROVED** [review]

**Pull request merged**